### ISSUE TITLE
FIgure out proper operational proof size weight

### ISSUE THREAD

**Issue by KiChjang** [paritytech/polkadot#6090]
Since migration to weights v2, we have set the proof size weight for the maximum block  as follows:

https://github.com/paritytech/polkadot/blob/3dbe51576a82c1297be7b43601dd97bc57693316/runtime/common/src/lib.rs#L72-L73

And we have also set up the reserved operational weights as follows:

https://github.com/paritytech/polkadot/blob/3dbe51576a82c1297be7b43601dd97bc57693316/runtime/common/src/lib.rs#L134-L141

The rational behind this is to ensure that we leave at least 25% of the block weight for operational class extrinsics. This is definitely true in the cacse where only reference time weights are measured.

It would be great if we can double-check that the rationale stated here still holds true for proof size weights.

**Comment by KiChjang** [comment]
Paging researchers from W3F here -- @burdges or @AlistarStewart, would any of you have ideas about this?

**Comment by bkchr** [comment]
Polkadot doesn't care about the proof size. So, this is not important at all here.

**Comment by KiChjang** [comment]
Yes, but we're not talking about _now_, we're talking about when we finally properly account for proof size weights.

**Comment by xlc** [comment]
Relaychain will never need to worry about proof size weights. However we do need an issue for system parachains, which could be used as a reference implementation for all other parachains. I suggest move this issue to cumulus.

**Comment by bkchr** [comment]
> Relaychain will never need to worry about proof size weights.

Exactly
